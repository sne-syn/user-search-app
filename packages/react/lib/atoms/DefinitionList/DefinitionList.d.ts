import { FC } from "react";
interface DefinitionListProps {
    list: {
        [key: string]: number | string;
    };
}
declare const DefinitionList: FC<DefinitionListProps>;
export default DefinitionList;
