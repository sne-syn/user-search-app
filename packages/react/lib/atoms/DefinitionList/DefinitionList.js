import React from 'react';

const DefinitionList = ({ list }) => {
    return (React.createElement("dl", null, Object.keys(list).map((key) => {
        return (React.createElement("div", { key: key },
            React.createElement("h2", null,
                key,
                ": ",
                list[key]),
            React.createElement("hr", null)));
    })));
};

export { DefinitionList as default };
//# sourceMappingURL=DefinitionList.js.map
