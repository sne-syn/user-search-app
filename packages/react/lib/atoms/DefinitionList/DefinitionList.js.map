{"version":3,"file":"DefinitionList.js","sources":[],"sourcesContent":[],"names":[],"mappings":";;;;;;;;;;;;;;;"}