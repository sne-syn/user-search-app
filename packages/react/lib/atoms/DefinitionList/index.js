export { default } from './DefinitionList.js';
//# sourceMappingURL=index.js.map
