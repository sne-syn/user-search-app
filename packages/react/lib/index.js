export { default as Avatar } from './atoms/Avatar/Avatar.js';
export { default as Button } from './atoms/Button/Button.js';
export { default as CardSubtitle } from './atoms/CardSubtitle/CardSubtitle.js';
export { default as CardTitle } from './atoms/CardTitle/CardTitle.js';
export { default as Icon } from './atoms/Icon/Icon.js';
export { default as Input } from './atoms/Input/Input.js';
export { default as Logo } from './atoms/Logo/Logo.js';
export { default as DateInfo } from './atoms/DateInfo/DateInfo.js';
export { default as DefinitionList } from './atoms/DefinitionList/DefinitionList.js';
export { default as ErrorMessage } from './atoms/ErrorMessage/ErrorMessage.js';
export { default as TextBlock } from './atoms/TextBlock/TextBlock.js';
export { default as ThemeToggler } from './atoms/ThemeToggler/ThemeToggler.js';
export { default as VisuallyHidden } from './atoms/VisuallyHidden/VisuallyHidden.js';
export { default as Wrapper } from './atoms/Wrapper/Wrapper.js';
//# sourceMappingURL=index.js.map
