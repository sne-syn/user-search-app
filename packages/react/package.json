{
  "name": "@user-search-app/react",
  "version": "1.0.0",
  "main": "lib/index.js",
  "license": "MIT",
  "scripts": {
    "build": "rollup -c",
    "dev": "yarn build --watch",
    "test": "jest --verbose",
    "test:watch": "yarn test --watch"
  },
  "devDependencies": {
    "@babel/preset-env": "^7.18.10",
    "@babel/preset-react": "^7.18.6",
    "@babel/preset-typescript": "^7.18.6",
    "@rollup/plugin-image": "^2.1.1",
    "@testing-library/jest-dom": "^5.16.5",
    "@testing-library/react": "^13.3.0",
    "@types/jest": "^28.1.6",
    "@types/react": "^18.0.15",
    "jest": "^28.1.3",
    "jest-environment-jsdom": "^28.1.3",
    "react": "^18.2.0",
    "rollup": "^2.77.0",
    "rollup-plugin-img": "^1.1.0",
    "rollup-plugin-typescript2": "^0.32.1",
    "typescript": "^4.7.4"
  }
}
