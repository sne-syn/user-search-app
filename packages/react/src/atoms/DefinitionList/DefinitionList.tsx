import React, { FC } from "react";

interface DefinitionListProps {
  list: { [key: string]: number | string };
}

const DefinitionList: FC<DefinitionListProps> = ({ list }) => {
  const capitalizeChar = (str: string): string =>
    str.charAt(0).toUpperCase() + str.slice(1);

  return (
    <dl className="dse-definitions" data-testid="definition-list">
      {Object.keys(list).map((key) => {
        return (
          <div
            key={key}
            className="dse-definition"
            data-testid="definition-item"
          >
            <dt className="dse-definition__term">{capitalizeChar(key)}</dt>
            <dd className="dse-definition__desc">{list[key]}</dd>
          </div>
        );
      })}
    </dl>
  );
};

export default DefinitionList;
