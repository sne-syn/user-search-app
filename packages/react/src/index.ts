import Avatar from "./atoms/Avatar";
import Button from "./atoms/Button";
import CardSubtitle from "./atoms/CardSubtitle";
import CardTitle from "./atoms/CardTitle";
import Input from "./atoms/Input";
import ListContent from "./atoms/ListContent";
import ListIcon from "./atoms/ListIcon";
import Logo from "./atoms/Logo";
import DateInfo from "./atoms/DateInfo";
import DefinitionList from "./atoms/DefinitionList";
import ErrorMessage from "./atoms/ErrorMessage";
import TextBlock from "./atoms/TextBlock";
import ThemeToggler from "./atoms/ThemeToggler";
import VisuallyHidden from "./atoms/VisuallyHidden";
import Wrapper from "./atoms/Wrapper";

export {
  Avatar,
  Button,
  CardSubtitle,
  CardTitle,
  Input,
  ListContent,
  ListIcon,
  Logo,
  DateInfo,
  DefinitionList,
  ErrorMessage,
  TextBlock,
  ThemeToggler,
  VisuallyHidden,
  Wrapper,
};
