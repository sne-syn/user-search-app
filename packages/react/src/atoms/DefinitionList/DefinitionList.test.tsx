import React from "react";
import DefinitionList from "./DefinitionList";
import { render, screen } from "@testing-library/react";

const mockList = { books: 8, followers: 3938, readers: 9 };

const renderComponent = () => {
  return render(<DefinitionList list={mockList} />);
};

describe("DefinitionList", () => {
  test("renders with date on human readable format", () => {
    renderComponent();
    expect(screen.getByTestId("definition-list")).toBeInTheDocument();
    expect(screen.getByTestId("definition-list")).toMatchSnapshot();
  });
  test("renders the right amount of definitions", () => {
    renderComponent();
    const definitionItems = screen.getAllByTestId("definition-item");
    expect(definitionItems).toHaveLength(Object.keys(mockList).length);
  });
});
