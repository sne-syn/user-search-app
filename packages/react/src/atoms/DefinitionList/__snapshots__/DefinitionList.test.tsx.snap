// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DefinitionList renders with date on human readable format 1`] = `
<dl
  class="dse-definitions"
  data-testid="definition-list"
>
  <div
    class="dse-definition"
    data-testid="definition-item"
  >
    <dt
      class="dse-definition__term"
    >
      Books
    </dt>
    <dd
      class="dse-definition__desc"
    >
      8
    </dd>
  </div>
  <div
    class="dse-definition"
    data-testid="definition-item"
  >
    <dt
      class="dse-definition__term"
    >
      Followers
    </dt>
    <dd
      class="dse-definition__desc"
    >
      3938
    </dd>
  </div>
  <div
    class="dse-definition"
    data-testid="definition-item"
  >
    <dt
      class="dse-definition__term"
    >
      Readers
    </dt>
    <dd
      class="dse-definition__desc"
    >
      9
    </dd>
  </div>
</dl>
`;
